### PR TITLE
Bump eventing to v1.7

### PR DESCRIPTION
**Pull request by ReToCode**
## Proposed Changes
- Bump eventing to 1.7 in `olm-catalog/serverless-operator/project.yaml` and `hack/update-deps.sh`
- Update dependencies with `make generated-files` and `./hack/update-deps.sh --upgrade`

Next try of partially doing https://github.com/openshift-knative/serverless-operator/pull/1888.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -171,10 +171,10 @@ require (
 
 replace (
 	// Knative forks.
-	knative.dev/eventing => github.com/openshift-knative/eventing v0.99.1-0.20221215170220-da8c25a653e7
+	knative.dev/eventing => github.com/openshift-knative/eventing v0.99.1-0.20221223064251-7859459225d6
 	knative.dev/eventing-kafka => github.com/openshift-knative/eventing-kafka v0.19.1-0.20221109100552-26eea5293bca
```
Comment (matzew) [inline go.mod:175]: In a separate PR I'd also update the `eventing-kafka` b/c of https://github.com/openshift-knative/eventing-kafka/commit/81b8c0ceb5408ff1d75f016fdd8e9f455b17ad4c

File: openshift-knative-operator/cmd/operator/kodata/knative-eventing/1.7/2-eventing-core.yaml [hunk 2]
```diff
@@ -4088,32 +4088,13 @@ spec:
...
-          timeoutSeconds: 5
-        readinessProbe:
-          httpGet:
-            path: /readiness
-            port: probes
-            scheme: HTTP
-          initialDelaySeconds: 20
-          periodSeconds: 10
-          timeoutSeconds: 5
+            - all
```
Comment (matzew) [inline openshift-knative-operator/cmd/operator/kodata/knative-eventing/1.7/2-eventing-core.yaml:4091]: we need to wait for the patch from @creydr

File: openshift-knative-operator/cmd/operator/kodata/knative-eventing/1.7/2-eventing-core.yaml [hunk 3]
```diff
@@ -4088,32 +4088,13 @@ spec:
           runAsNonRoot: true
           capabilities:
             drop:
-            - ALL
-
-        livenessProbe:
```
Comment (matzew) [inline openshift-knative-operator/cmd/operator/kodata/knative-eventing/1.7/2-eventing-core.yaml:4093]: Oh, I need to check what's this, actually

File: go.mod (hunk 1 above)
Comment (ReToCode) [reply go.mod:175]: Yeah lets just update the manifests, then all the go deps in one go (as there are conflicts between eventing & knative-operator, if done separately).

File: openshift-knative-operator/cmd/operator/kodata/knative-eventing/1.7/2-eventing-core.yaml (hunk 3 above)
Comment (matzew) [reply openshift-knative-operator/cmd/operator/kodata/knative-eventing/1.7/2-eventing-core.yaml:4093]: @ReToCode fixed via: https://github.com/openshift-knative/eventing/pull/60

Comment (matzew) [reply openshift-knative-operator/cmd/operator/kodata/knative-eventing/1.7/2-eventing-core.yaml:4093]: I updated this part already

File: olm-catalog/serverless-operator/project.yaml [hunk 4]
```diff
@@ -37,9 +37,9 @@ dependencies:
   net_istio_artifacts_branch: release-1.7
   maistra: 2.1.0
 
-  eventing: knative-v1.6
+  eventing: knative-v1.7
```
Comment (matzew) [inline olm-catalog/serverless-operator/project.yaml:40]: Our backports are now merged...

Comment (ReToCode) [reply olm-catalog/serverless-operator/project.yaml:40]: I'm trying https://github.com/openshift-knative/serverless-operator/pull/1888 first. If this doesn't work out, I'll continue here.

**Review by matzew — APPROVED** [review]
/lgtm
/approve

Feel free to unhold

**Comment by matzew** [comment]
/close

replaced by #1888  (was more of a test)

**Pull request closed without merging**